torchrec/distributed/tests/test_model_parallel.py

```diff
@@ -335,22 +340,54 @@ def test_sharding_gloo_dp(
             backend="gloo",
         )
 
-    def test_sharding_ebc_as_top_level(self) -> None:
-        os.environ["RANK"] = "0"
-        os.environ["WORLD_SIZE"] = "1"
-        os.environ["LOCAL_WORLD_SIZE"] = "1"
-        os.environ["MASTER_ADDR"] = str("localhost")
-        os.environ["MASTER_PORT"] = str(get_free_port())
-        os.environ["NCCL_SOCKET_IFNAME"] = "lo"
+    @unittest.skipIf(
+        torch.cuda.device_count() <= 1,
+        "Not enough GPUs, this test requires at least two GPUs",
+    )
+    def test_top_lovel_shardable_module_is_sharded_nccl(self) -> None:
+        backend = "nccl"
+        device = torch.device("cuda:0")
+        pg = init_distributed_single_host(
+            rank=0,
+            world_size=1,
+            backend=backend,
+            local_size=1,
+        )
 
-        if torch.cuda.is_available():
-            curr_device = torch.device("cuda:0")
-            torch.cuda.set_device(curr_device)
-            backend = "nccl"
-        else:
-            curr_device = torch.device("cpu")
-            backend = "gloo"
-        dist.init_process_group(backend=backend)
+        embedding_dim = 128
+        num_embeddings = 256
+        ebc = EmbeddingBagCollection(
+            device=torch.device("meta"),
+            tables=[
+                EmbeddingBagConfig(
+                    name="large_table",
+                    embedding_dim=embedding_dim,
+                    num_embeddings=num_embeddings,
+                    feature_names=["my_feature"],
+                    pooling=PoolingType.SUM,
+                ),
+            ],
+        )
+
+        model = DistributedModelParallel(ebc, device=device)
+
+        self.assertTrue(isinstance(model.module, ShardedEmbeddingBagCollection))
+
+        if _INTRA_PG is not None:
```

Comment:
btw, these PGs won't be created in this test (only for hierarchical sharding style )

Reply:
Thanks for teaching me!
I gained a deeper understanding.